### PR TITLE
User data fixes

### PR DESCRIPTION
**Pull request by Dudar99**

### CODE REVIEW

File: car-statistics/app/routers/user_data.py [hunk 1]
```diff
@@ -0,0 +1,13 @@
+from app import app
```
Comment (ssaporylo) [inline car-statistics/app/routers/user_data.py]: Do you need it?

File: car-statistics/app/services/user_data_collection.py [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    return files
+
+
+def get_user_filters(user_id):
+    """
+
+    :param user_id:
+    :return:
+    """
+    filters_ids = [dts['filter_id'] for dts in get_user_datasets(user_id) if type(dts['filter_id']) == int]
```
Comment (ssaporylo) [inline car-statistics/app/services/user_data_collection.py]: Why do you need here type(dts['filter_id']) == int?

File: car-statistics/app/__init__.py [hunk 3]
```diff
@@ -52,4 +52,6 @@
     filter,
     file_data,
     datasets,
+    user_data,
```
Comment (ssaporylo) [inline car-statistics/app/__init__.py]: alphabetical please

File: car-statistics/app/routers/delete_file.py [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+
+from flask import session
+from flask_login import login_required
+from app.models import File, Dataset
+from app.services.utils import user_dir, ext_free, delete_files_from_dir
+import os
+@app.route('/api/delete_file/<int:file_id>')
+@login_required
+def delete_file(file_id):
+
```
Comment (ssaporylo) [inline car-statistics/app/routers/delete_file.py]: Doc string please

File: car-statistics/app/routers/delete_file.py [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+    user_id = int(session.get('user_id', 0))
```
Comment (ssaporylo) [inline car-statistics/app/routers/delete_file.py]: Why default 0?

File: car-statistics/app/routers/delete_file.py [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+    dataset = Dataset.query.filter(Dataset.user_id == user_id).first()
+    file = File.query.filter(File.id == file_id).first()
+    if not file:
+        return json.dumps({'message': 'file does not exist'}), 404
```
Comment (ssaporylo) [inline car-statistics/app/routers/delete_file.py]: 404 use constant instead, please

File: car-statistics/app/routers/delete_file.py [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+    if dataset.user_id != user_id:
```
Comment (ssaporylo) [inline car-statistics/app/routers/delete_file.py]: Case dataset None?

File: car-statistics/app/services/utils.py [hunk 8]
```diff
@@ -96,7 +95,13 @@ def attributes(file_path):
     :param file_path: file which will be added to DB
     :return: json with file attributes
     """
-    attrbts = {'loaded': 'new'}
+    attrbts = dict()
+    attrbts['name'] = 'name'
+    file_size_MB = os.path.getsize(file_path)/(1024*1024.0) #in MBytes
```
Comment (ssaporylo) [inline car-statistics/app/services/utils.py]: It should be in utils to reuse it

File: car-statistics/app/services/utils.py [hunk 9]
```diff
@@ -96,7 +95,13 @@ def attributes(file_path):
...
+    attrbts['size'] = math.ceil(file_size_MB*100)/100       # round 2 decimals after poin
+    attrbts['rows'] = 0
+    attrbts['date'] = time.time()
```
Comment (ssaporylo) [inline car-statistics/app/services/utils.py]: You should use same approach https://github.com/lv336python/car-statistics/blob/master/car-statistics/app/models/files.py#L54 here for datetime generation used datetime module. And it should be in utils to provide it

File: car-statistics/app/services/utils.py [hunk 10]
```diff
@@ -172,3 +177,11 @@ def serialized_file(file):
     """
     file_pth = ext_free(file)
     return (f'{file_pth}.pkl')
+
+def delete_files_from_dir(dir, file):
+
```
Comment (ssaporylo) [inline car-statistics/app/services/utils.py]: Doc string please

**Comment by ssaporylo** [comment]
Do you run linter? Pls resolve conflicts

File: car-statistics/app/static/src/app/_services/user-data.service.ts [hunk 11]
```diff
@@ -0,0 +1,39 @@
...
+  }
+
+  getUserData() {
+    return this.http
+        .get('api/userdata')
+        .catch(this.handleError);
+
+  }
+
+  getUserFiles(data) {
```
Comment (ssaporylo) [inline car-statistics/app/static/src/app/_services/user-data.service.ts]: I don't recomend to usie combined name convention

File: car-statistics/app/static/src/app/models/data.ts [hunk 12]
```diff
@@ -0,0 +1,32 @@
+export class UserData {
+    constructor(public user_files: any,
```
Comment (ssaporylo) [inline car-statistics/app/static/src/app/models/data.ts]: Bad practice to use any here

Comment (ssaporylo) [reply car-statistics/app/static/src/app/models/data.ts]: All these models whould be in separate file

File: car-statistics/app/static/src/app/user-data/user-files/user-file/user-file.component.css [hunk 13]
```diff
@@ -0,0 +1,3 @@
+.file-stat{
+    margin-left: 10%;
+}
```
Comment (ssaporylo) [inline car-statistics/app/static/src/app/user-data/user-files/user-file/user-file.component.css]: Pls fix it

File: car-statistics/app/static/src/app/user-data/user-files/user-file/user-file.component.ts [hunk 14]
```diff
@@ -0,0 +1,40 @@
...
+
+
+@Component({
+  selector: 'app-user-file',
+  templateUrl: './user-file.component.html',
+  styleUrls: ['./user-file.component.css']
+})
+export class UserFileComponent implements OnInit {
+  @Input() userfile: File;
+  @Input() i; //for files numeration
```
Comment (ssaporylo) [inline car-statistics/app/static/src/app/user-data/user-files/user-file/user-file.component.ts]: pls change 'i' no need comment here

File: car-statistics/app/static/src/app/user-data/user-files/user-file/user-file.component.ts [hunk 15]
```diff
@@ -0,0 +1,40 @@
...
+  @Input() dataset_id; //for files numeration
+
+  showDatasetStat : boolean = false;
+  fileDeleted :boolean = false;
+
+  getStatDataset(){
```
Comment (ssaporylo) [inline car-statistics/app/static/src/app/user-data/user-files/user-file/user-file.component.ts]: Why don't join getStatDataset() and hideStatDataset()?

File: car-statistics/app/static/src/app/user-data/user-files/user-file/user-file.component.ts [hunk 16]
```diff
@@ -0,0 +1,40 @@
...
+  }
+  hideDatasetStat(){
+    this.showDatasetStat = false;
+  }
+
+  deleteFile(fileId){
+    this.file.deleteUserFile(fileId).subscribe(
+        res =>{
+          this.fileDeleted = true;
+        }
```
Comment (ssaporylo) [inline car-statistics/app/static/src/app/user-data/user-files/user-file/user-file.component.ts]: What do you think about handle errors here?

File: car-statistics/app/static/src/app/user-data/user-filters/user-filter/user-filter.component.ts [hunk 17]
```diff
@@ -0,0 +1,16 @@
...
+import {Filter} from '../../../models/data';
+
+@Component({
+  selector: 'app-user-filter',
+  templateUrl: './user-filter.component.html',
+  styleUrls: ['./user-filter.component.css']
+})
+export class UserFilterComponent implements OnInit {
+  @Input() userfilter: Filter;
+  constructor() { }
```
Comment (ssaporylo) [inline car-statistics/app/static/src/app/user-data/user-filters/user-filter/user-filter.component.ts]: Why do you need empty constructor here?

File: car-statistics/app/static/src/app/user-data/user-filters/user-filter/user-filter.component.ts [hunk 18]
```diff
@@ -0,0 +1,16 @@
...
+
+  ngOnInit() {
```
Comment (ssaporylo) [inline car-statistics/app/static/src/app/user-data/user-filters/user-filter/user-filter.component.ts]: Do we need empty ngOnInit() here?

File: car-statistics/app/static/src/app/user-file.service.ts [hunk 19]
```diff
@@ -0,0 +1,17 @@
+import { Injectable } from '@angular/core';
+import {HttpClient} from '@angular/common/http'
+import {Observable} from "rxjs";
+@Injectable({
+  providedIn: 'root'
+})
+export class UserFileService {
+  deleteUserFileUrl = '/api/delete_file'
```
Comment (ssaporylo) [inline car-statistics/app/static/src/app/user-file.service.ts]: Should be in Constants file

File: car-statistics/app/routers/user_data.py (hunk 1 above)
Comment (Dudar99) [reply car-statistics/app/routers/user_data.py]: DONE, app needed for routs creation

File: car-statistics/app/services/utils.py (hunk 10 above)
Comment (Dudar99) [reply car-statistics/app/services/utils.py]: DONE

File: car-statistics/app/static/src/app/user-data/user-files/user-file/user-file.component.ts (hunk 16 above)
Comment (Dudar99) [reply car-statistics/app/static/src/app/user-data/user-files/user-file/user-file.component.ts]: Done

**Pull request merged**